torch/_refs/__init__.py

```diff
@@ -6358,12 +6358,17 @@ def _infer_scalar_type(obj):
 
 # Analogous to recursive_store
 # xref: recursive_store in torch/csrc/utils/tensor_new.cpp
-def _recursive_build(scalarType: torch.dtype, obj: TensorOrNumberLikeType):
+def _recursive_build(
+    scalarType: torch.dtype, obj: Union[TensorOrNumberLikeType, TensorSequenceType]
+):
     if isinstance(obj, Tensor) and obj.ndim <= 1:
         return obj.detach().to(dtype=scalarType, device="cpu", copy=True).view(())
+    elif isinstance(obj, Tensor):
+        return obj.detach().to(dtype=scalarType, device="cpu", copy=True)
```

Comment:
I'm not sure you'll necessarily end up with the right shape without an appropriate view.  Specifically:

```
>>> torch.tensor([torch.tensor(0)])
tensor([0])
>>> torch.tensor([torch.tensor([0])])
tensor([0])
>>> torch.tensor([torch.tensor([[0]])])
tensor([0])
>>> torch.tensor([torch.tensor([[[0]]])])
tensor([0])
```